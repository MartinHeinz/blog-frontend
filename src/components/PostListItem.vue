<template>
    <li class="post-item">
        <div class="meta"><time :datetime="published" itemprop="datePublished">{{ published | formatDate }}</time></div>
        <span>
            <router-link :to="url">{{value}}</router-link>
        </span>
    </li>
</template>

<script>
export default {
    name: 'PostListItem',
    props: {
        published: String,
        url: String,
        value: String,
    },
};
</script>

<style>
    .post-list .post-item .meta {
        display: block;
        margin-right: 16px;
        min-width: 100px;
        color: #666;
        font-size: 14px;
    }
</style>
