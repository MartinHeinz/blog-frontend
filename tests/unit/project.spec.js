import { createLocalVue, mount } from '@vue/test-utils';
import PostListItem from '@/components/Project.vue';
import Router from 'vue-router';

describe('Project.vue', () => {
    it('renders correcly', () => {
        const localVue = createLocalVue();
        const router = new Router({ mode: 'history' });
        localVue.use(Router);

        const name = 'My Project';
        const url = 'https://github.com/something';
        const description = 'This is description of this project...';
        const src = 'https://via.placeholder.com/150';
        const tags = [
            {
                name: 'Golang',
                url: 'golang',
            },
            {
                name: 'Vue',
                url: 'vue',
            },
        ];

        const wrapper = mount(PostListItem, {
            localVue,
            router,
            propsData: {
                name,
                url,
                description,
                src,
                tags,
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
