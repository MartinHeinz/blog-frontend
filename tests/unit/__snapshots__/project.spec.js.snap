// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project.vue renders correcly 1`] = `
<article class="project-item">
  <figure class="left"><a href="https://github.com/something" class="no-underline"><img src="https://via.placeholder.com/150" alt="My Project" height="150" width="250"></a></figure>
  <section class="right">
    <h2>My Project</h2>
    <p>This is description of this project...</p>
    <div class="project-tag">
      <v-icon>fas fa-tag</v-icon> <a href="/tags/golang/" class="tag-link no-underline">Golang</a><a href="/tags/vue/" class="tag-link no-underline">Vue</a>
    </div>
  </section>
</article>
`;
